src/main/java/elucent/rootsclassic/recipe/RitualRecipe.java

```diff
@@ -0,0 +1,193 @@
+package elucent.rootsclassic.recipe;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import elucent.rootsclassic.Const;
+import elucent.rootsclassic.block.brazier.BrazierBlockEntity;
+import elucent.rootsclassic.registry.RootsRecipes;
+import elucent.rootsclassic.ritual.RitualBaseRegistry;
+import elucent.rootsclassic.ritual.RitualEffect;
+import elucent.rootsclassic.ritual.RitualPillars;
+import elucent.rootsclassic.ritual.rituals.RitualCrafting;
+import elucent.rootsclassic.util.RootsUtil;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.NonNullList;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.world.Container;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.*;
```

Comment:
Wildcard import

Reply:
> 

Not really, my brain has not fully moved on to 1.19 yet and the default branch of roots was trunk/1.18, I did not realize their was a branch for 1.19 already